pdns/sstuff.hh

```diff
@@ -127,6 +128,15 @@ public:
     }
   }
 
+  void setFastOpenConnect()
+  {
+#ifdef TCP_FASTOPEN_CONNECT
+    int on = 1;
+    if (setsockopt(d_socket, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, &on, sizeof(on)) < 0)
+      throw NetworkError("While setting TCP_FASTOPEN_CONNECT: " + stringerror());
```

Comment:
Do we want to fall on our sword on that? I'm pretty sure we will have some cases where `TCP_FASTOPEN_CONNECT` was defined at compile time but it not available at run-time. So it might make sense to catch that exception in the callers?

Reply:
or log and continue, as we do for the listening case.

Reply:
Agreed, provided that we catch and log in the caller.